cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,73 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
+import homePageObject from '../support/pages/home.pageObject';
+import settingsPageObject from '../support/pages/settings.pageObject';
+import SignInPageObject from '../support/pages/signIn.pageObject';
 
-describe('Settings page', () => {
-  before(() => {
+const homePage = new homePageObject();
+const settingsPage = new settingsPageObject();
+const signInPage = new SignInPageObject();
 
-  });
+const userData = {
+  username: 'riot',
+  email: 'riot@qa.team',
+  password: '12345Qwert!',
+}
 
-  beforeEach(() => {
+describe('Settings page', () => {
+  let user;
 
+  beforeEach(() => {
+    homePage.clearDatabase();
+    homePage.visit();
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
+    settingsPage.loggedUser();
+    settingsPage.visit();
   });
 
   it('should provide an ability to update username', () => {
-
+    settingsPage.updateUsername(user.username);
+    settingsPage.clickSubmitButton();
+    homePage.newUsernameLink(user.username);
   });
 
   it('should provide an ability to update bio', () => {
-
+    settingsPage.updateBio(user.bio);
+    settingsPage.clickSubmitButton();
+    homePage.clickSettingsLink();
+    settingsPage.assertBio(user.bio);
   });
 
   it('should provide an ability to update an email', () => {
-
+    settingsPage.updateEmail(user.email);
+    settingsPage.clickSubmitButton();
+    homePage.clickSettingsLink();
+    settingsPage.logOut();
+    settingsPage.assertLogOut();
+    signInPage.visit();
+    signInPage.emailField(user.email);
+    signInPage.passwordField(userData.password);
+    signInPage.signInBtn();
+    homePage.usernameLink(userData.username);
   });
 
   it('should provide an ability to update password', () => {
-
+    settingsPage.updatePassword(user.password);
+    settingsPage.clickSubmitButton();
+    homePage.clickSettingsLink();
+    settingsPage.logOut();
+    settingsPage.assertLogOut();
+    signInPage.visit();
+    signInPage.emailField(userData.email);
+    signInPage.passwordField(user.password);
+    signInPage.signInBtn();
+    homePage.usernameLink(userData.username);
```

Comment:
a lot of duplicated code. create more methods and reuse them to avoid duplicating.
use the rule if a line repeats 2 or more times it should be another method